diff --git a/src/app/app.module.ts b/src/app/app.module.ts
index f28974676e..717d7ca646 100644
--- a/src/app/app.module.ts
+++ b/src/app/app.module.ts
@@ -151,6 +151,8 @@ import { AddonQtypeModule } from '@addon/qtype/qtype.module';
 import { AddonStorageManagerModule } from '@addon/storagemanager/storagemanager.module';
 import { AddonFilterModule } from '@addon/filter/filter.module';

+import { CustomModule } from '../custom/custom.module';
+
 // For translate loader. AoT requires an exported function for factories.
 export function createTranslateLoader(http: HttpClient): TranslateHttpLoader {
     return new TranslateHttpLoader(http, './assets/lang/', '.json');
@@ -294,7 +296,8 @@ export const WP_PROVIDER: any = null;
         AddonQbehaviourModule,
         AddonQtypeModule,
         AddonStorageManagerModule,
-        AddonFilterModule
+        AddonFilterModule,
+        CustomModule,
     ],
     bootstrap: [IonicApp],
     entryComponents: [
@@ -338,7 +341,8 @@ export const WP_PROVIDER: any = null;
         {provide: COMPILER_OPTIONS, useValue: {}, multi: true},
         {provide: JitCompilerFactory, useClass: JitCompilerFactory, deps: [COMPILER_OPTIONS]},
         {provide: LocationStrategy, useClass: MockLocationStrategy},
-    ]
+        ...CustomModule.rootProviders,
+    ],
 })
 export class AppModule {
     constructor(platform: Platform, initDelegate: CoreInitDelegate, updateManager: CoreUpdateManagerProvider, config: Config,
